### PR TITLE
Floating Action Button Disappears when list of topic is scrolled. #117

### PR DESCRIPTION
**Pull request by JubrilEdu**
Connected issue: https://github.com/mgolokhov/dodroid/issues/117
- This PR fixes the issue with FAB Disappearing.
- A listener was added to the hide method, it seems the hide method does some layout actions that remove the view. 
Cc @mgolokhov

### CODE REVIEW

File: app/src/main/java/doit/study/droid/Utils/Sound.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package doit.study.droid.utils;
+package doit.study.droid.Utils;
```
Comment (mgolokhov) [inline app/src/main/java/doit/study/droid/Utils/Sound.java]: Please revert to the initial lower case.
Kotlin's naming conversions https://kotlinlang.org/docs/reference/coding-conventions.html#naming-rules

File: app/src/main/java/doit/study/droid/activities/DrawerBaseActivity.kt [hunk 2]
```diff
@@ -5,6 +5,7 @@ import android.content.Intent
 import android.content.res.Configuration
 import android.net.Uri
 import android.os.Bundle
+import android.provider.MediaStore.getVersion
```
Comment (mgolokhov) [inline app/src/main/java/doit/study/droid/activities/DrawerBaseActivity.kt]: Wrong import, should be `import doit.study.droid.utils.Distribution.getVersion`

File: app/src/main/java/doit/study/droid/fragments/TopicsChooserFragment.kt [hunk 3]
```diff
@@ -41,6 +43,7 @@ class TopicsChooserFragment : Fragment(), LoaderManager.LoaderCallbacks<Cursor>,
     // loaders resets state, have to save in var
     private var savedRecyclerLayoutState: Parcelable? = null
 
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setHasOptionsMenu(true)
```
Comment (mgolokhov) [inline app/src/main/java/doit/study/droid/fragments/TopicsChooserFragment.kt]: Revert file to the initial state, please.

File: app/src/main/java/doit/study/droid/views/ScrollAwareFABBehavior.kt [hunk 4]
```diff
@@ -33,10 +36,21 @@ class ScrollAwareFABBehavior : FloatingActionButton.Behavior {
                 dyUnconsumed)
 
         if (dyConsumed > 0 && child.visibility == View.VISIBLE) {
-            child.hide()
-        } else if (dyConsumed < 0 && child.visibility != View.VISIBLE) {
+            child.hide(object : FloatingActionButton.OnVisibilityChangedListener() {
+                override fun onShown(fab: FloatingActionButton?) {
+                    super.onShown(fab)
+                }
```
Comment (mgolokhov) [inline app/src/main/java/doit/study/droid/views/ScrollAwareFABBehavior.kt]: You override nothing, `onShown()` can be removed.

File: app/src/main/res/layout/fragment_topics_chooser.xml [hunk 5]
```diff
@@ -12,25 +12,21 @@
...
         android:layout_gravity="bottom|right"
         app:layout_anchorGravity="bottom|right"
         app:pressedTranslationZ="12dp"
         android:layout_margin="16dp"
-        app:useCompatPadding="true"
-        app:layout_behavior="doit.study.droid.views.ScrollAwareFABBehavior"
-        />
+        app:useCompatPadding="true"/>
 
 </androidx.coordinatorlayout.widget.CoordinatorLayout>
```
Comment (mgolokhov) [inline app/src/main/res/layout/fragment_topics_chooser.xml]: The file can be reverted to the initial state.

**Comment by mgolokhov** [comment]
@djubreel thank you for the pull request.
After minor fixes, it can be merged.
Please check the comments.

**Review by mgolokhov — CHANGES_REQUESTED** [review]

**Comment by JubrilEdu** [comment]
Gone through the comments and made adjustments.

**Comment by mgolokhov** [comment]
@djubreel 
Just a final step to rename folder Utils -> utils

**Comment by JubrilEdu** [comment]
@mgolokhov I've done that, you can verify that via the import statements.

**Comment by qps-ua-mgolokhov** [comment]
@djubreel https://github.com/djubreel/dodroid/tree/master/app/src/main/java/doit/study/droid
Screenshot 2019-10-22 at 17 05 28

**Comment by mgolokhov** [comment]
@djubreel https://github.com/djubreel/dodroid/tree/master/app/src/main/java/doit/study/droid
67294207-33cd5800-f4ee-11e9-8315-2d8192d7503e

**Comment by JubrilEdu** [comment]
I have no idea why. It's lowercased on my local.
Check your repo, it's the same there.

**Pull request merged**